crates/sui-indexer-alt-framework/src/ingestion/rpc_client.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::ingestion::client::{FetchData, FetchError, FetchResult, IngestionClientTrait};
+use anyhow::anyhow;
+use sui_rpc_api::Client as RpcClient;
+
+#[async_trait::async_trait]
+impl IngestionClientTrait for RpcClient {
+    async fn fetch(&self, checkpoint: u64) -> FetchResult {
+        let data = self.get_full_checkpoint(checkpoint).await.map_err(|e| {
+            if e.message().contains("not found") {
```

Comment:
Is there a more structured way to identify that a checkpoint is not found based on the response @bmwill ?

Reply:
The errors are a bit of a mess still and i'm actively working on this as we speak. right now i think the gRPC Code::Unknown is always returned (which is not great)

Reply:
There is a `Code::NOT_FOUND` though which will eventually be returned here (hopefully by the next release)

Reply:
ok https://github.com/MystenLabs/sui/pull/21163 should address this and allow you to look at the error code directly